components/calendar/calendar.js

```diff
@@ -0,0 +1,480 @@
+	}
+
+	static get resources() {
+		return {
+			'ar': { show: 'Show {month} ' },
+			'en': { show: 'Show {month}' },
+			'es': { show: 'Show {month} ' },
+			'fr': { show: 'Show {month} ' },
+			'ja': { show: 'Show {month} ' },
+			'ko': { show: 'Show {month} ' },
+			'nl': { show: 'Show {month} ' },
+			'pt': { show: 'Show {month} ' },
+			'sv': { show: 'Show {month} ' },
+			'tr': { show: 'Show {month} ' },
+			'zh': { show: 'Show {month} ' },
+			'zh-tw': { show: 'Show {month} ' }
+		};
+	}
+
+	constructor() {
+		super();
+
+		this._descriptor = getDateTimeDescriptor();
+
+		this._daysOfWeekIndex = [];
+		for (let i = this._descriptor.calendar.firstDayOfWeek; i < this._descriptor.calendar.firstDayOfWeek + daysInWeek; i++) {
+			this._daysOfWeekIndex.push(i % daysInWeek);
+		}
+
+		this._dialogLabelId = getUniqueId();
+	}
+
+	firstUpdated(changedProperties) {
+		super.firstUpdated(changedProperties);
+
+		this.selectedValue = this.selectedValue ? new Date(this.selectedValue) : null;
+		this._today = new Date();
+		this._focusDate = this.selectedValue ? new Date(this.selectedValue.getFullYear(), this.selectedValue.getMonth(), this.selectedValue.getDate()) : new Date(this._today.getFullYear(), this._today.getMonth(), 1);
+		this._shownMonth = this.selectedValue ? this.selectedValue.getMonth() : this._today.getMonth();
+		this._shownYear = this.selectedValue ? this.selectedValue.getFullYear() : this._today.getFullYear();
+	}
+
+	updated(changedProperties) {
+		super.updated(changedProperties);
+
+		changedProperties.forEach((oldVal, prop) => {
+			if (prop === '_shownMonth') {
+				this._focusDateAddFocus();
+			}
+		});
+	}
+
+	render() {
+		const weekdayHeaders = this._daysOfWeekIndex.map((index) => html`
+			<th role="columnheader" abbr="${this._descriptor.calendar.days.long[index]}">
+				<abbr class="d2l-body-small" title="${this._descriptor.calendar.days.long[index]}">
+					${this._descriptor.calendar.days.short[index]}
+				</abbr>
+			</th>
+		`);
+
+		const dates = getDatesInMonthArray(this._shownMonth, this._shownYear, this._descriptor.calendar.firstDayOfWeek);
+		const dayRows = dates.map((week, index) => {
+			const weekHtml = week.map((day) => {
+				const classes = {
+					'd2l-calendar-date': true,
+					'd2l-calendar-date-first-row': index === 0 ? true : false,
+					'd2l-calendar-date-selected': checkIfDatesEqual(day, this.selectedValue),
+					'd2l-calendar-date-today': checkIfDatesEqual(day, this._today)
+				};
+				const focused = checkIfDatesEqual(day, this._focusDate);
+				return html`
+					<td>
+						<div
+							@click="${this._onDateSelected}"
+							@keydown="${this._onKeyDown}"
+							class=${classMap(classes)}
+							data-date=${day.getDate()}
+							data-month=${day.getMonth()}
+							data-year=${day.getFullYear()}
+							tabindex=${focused ? '0' : '-1'}>
+							${day.getDate()}
+						</div>
+					</td>`;
+			});
+
+			return html`
+				<tr>${weekHtml}</tr>
+			`;
```

Comment:
Super-nit: I think for really small snippets like this we could just put it all on one line